test/runtests.py

```diff
@@ -506,7 +547,7 @@ def timeout_func(timeout_data):
             return self._show_failed(timedout=True, **fail_args)
 
         # check ch failed
-        if exit_code != 0:
+        if exit_code != 0 and binary_name_noext == 'ch':
```

Comment:
Why would this be specific to ch?

Reply:
I believe that an uncaught exception in jshost causes us to exit with a non-zero error code, but in ch it causes us to exit with a 0 error code. We have some tests where the baseline includes the error message at the end. We may want to normalize this behavior in the future, but that's out-of-scope for this change.